File: task3/main.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Введите число: ");
        int n = int.Parse(Console.ReadLine());
        int count = 0;

        for (int i = 0; i < n; i++)
        {
            int num = int.Parse(Console.ReadLine());
            int lastDigit = num % 10;
            num /= 10;
            bool sameDigit = true;

            while (num > 0)
            {
                int currentDigit = num % 10;
                if (currentDigit != lastDigit)
                {
                    sameDigit = false;
                    break;
                }
                num /= 10;
            }

            if (sameDigit)
                count++;
        }

        Console.WriteLine($"Количество: {count}");
    }
}

File: task3/main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите число: ");
        int n = scanner.nextInt();
        int count = 0;

        for (int i = 0; i < n; i++) {
            int num = scanner.nextInt();
            int lastDigit = num % 10;
            num /= 10;
            boolean sameDigit = true;

            while (num > 0) {
                int currentDigit = num % 10;
                if (currentDigit != lastDigit) {
                    sameDigit = false;
                    break;
                }
                num /= 10;
            }

            if (sameDigit) {
                count++;
            }
        }

        System.out.println("Количество: " + count);
    }
}
